### PR TITLE
[Communication] - PhoneNumbers - Rename API methods

### PR DESCRIPTION
**Pull request by pavelprystinka**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by minnieliu** [comment]
We should also update the changelog

File: sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumbersModelFactory.cs [hunk 1]
```diff
@@ -21,9 +21,9 @@ public static class PhoneNumbersModelFactory
         /// <param name="purchaseDate"> The date and time that the phone number was purchased. </param>
         /// <param name="cost"> The incurred cost for a single phone number. </param>
 #pragma warning disable CA1054 // URI-like parameters should not be strings
-        public static AcquiredPhoneNumber AcquiredPhoneNumber(string id, string phoneNumber, string countryCode, PhoneNumberType phoneNumberType, PhoneNumberCapabilities capabilities, PhoneNumberAssignmentType assignmentType, DateTimeOffset purchaseDate, PhoneNumberCost cost)
+        public static PurchasedPhoneNumber AcquiredPhoneNumber(string id, string phoneNumber, string countryCode, PhoneNumberType phoneNumberType, PhoneNumberCapabilities capabilities, PhoneNumberAssignmentType assignmentType, DateTimeOffset purchaseDate, PhoneNumberCost cost)
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumbersModelFactory.cs:24]: Still have instance of AcquiredPhoneNumber on line 24

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 2]
```diff
@@ -197,12 +197,12 @@ public virtual UpdatePhoneNumberCapabilitiesOperation StartUpdateCapabilities(st
             }
         }
 
-        /// <summary> Gets the details of the given acquired phone number. </summary>
-        /// <param name="phoneNumber"> The acquired phone number whose details are to be fetched in E.164 format, e.g. +14255550123. </param>
+        /// <summary> Gets the details of the given purchased phone number. </summary>
+        /// <param name="phoneNumber"> The acquired phone number whose details are to be fetched in E.164 format, e.g. +11234567890. </param>
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:201]: Lets change the e.g. to `14255550123` this is an actual invalid number we are using across langs

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs [hunk 3]
```diff
@@ -215,12 +215,12 @@ public virtual async Task<Response<AcquiredPhoneNumber>> GetPhoneNumberAsync(str
...
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs:219]: Same here `+14255550123`

File: sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumbersModelFactory.cs (hunk 1 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumbersModelFactory.cs]: Fixed

File: sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs (hunk 2 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.PhoneNumbers/src/PhoneNumbersClient.cs]: Fixed

**Review by minnieliu — APPROVED** [review]

**Pull request merged**